src/java.base/share/classes/java/util/zip/ZipInputStream.java

```diff
@@ -191,6 +192,204 @@ public int available() throws IOException {
         }
     }
 
+    /**
+     * Reads a byte of uncompressed data from the input stream for the current
+     * ZIP entry. This method will block until enough input is available for
+     * decompression.
+     * @return the byte read, or -1 if end of compressed input is reached
+     * @throws    IOException if an I/O error has occurred
+     */
+    @Override
+    public int read() throws IOException {
+        return super.read();
+    }
+
+    /**
+     * Reads all remaining bytes from the input stream for the current ZIP entry.
+     * This method blocks until all remaining bytes have been read and end of
+     * stream is detected, or an exception is thrown. This method does not close
```

Comment:
Hello Lance, the part where it says that it reads the current ZIP entry sounds fine. However, this also states that it will wait until the end of stream is detected. Should we make it clearer by changing this to say `This method blocks until all remaining bytes of the entry have been read or an exception is thrown`?

Reply:
Similar comment for the other `readNBytes` methods.

Reply:
Hi Jai,

Thank you for your input above.  I had thought about that but decided it would be best to keep the changes at a minimum from InputStream and felt given the change of the first sentence the rest of the wording is in regards to the current ZIP Entry